### PR TITLE
migrate analyses vdi ids

### PR DESCRIPTION
**Pull request by dmgaldi**
## Overview
* Migrate VDI entity and variable IDs in analyses
* It's a little bit hacky, so feel free to propose alternative approaches.

## Testing
* Ran with write disabled and inspected output.
* Will test once migration is done in QA.

### CODE REVIEW

File: Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java [hunk 1]
```diff
@@ -1,89 +1,153 @@
...
+
+    LOG.info("Analysis descriptor before migration: " + nextRow.getDescriptor());
+    String descriptor = nextRow.getDescriptor().toString();
+
+    // Find all variable IDs.
+    final Set<String> legacyVariableIds = VAR_ID_PATTERN.matcher(descriptor).results()
+        .map(match -> match.group(1))
+        .collect(Collectors.toSet());
+
+    final String entityId = ENTITY_ID_PATTERN.matcher(descriptor).results()
```
Comment (ryanrdoherty) [inline Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java:114]: Are we sure there's only one here?  We could easily have a config with >1 variable spec, each of which might contain a different entity ID, especially across different visualizations.

File: Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java [hunk 2]
```diff
@@ -1,89 +1,153 @@
...
+    }
+
+    // Replace all variable IDs with value converted from legacy variable ID.
+    for (String legacyVariableId: legacyVariableIds) {
+      descriptor = descriptor.replaceAll(legacyVariableId, convertToVdiId(legacyVariableId));
+    }
 
-    return new TableRowInterfaces.RowResult<>(nextRow)
+    // Create a copy with just the dataset ID updated to VDI counterpart.
+    AnalysisRow out = new AnalysisRow(nextRow.getAnalysisId(), vdiDatasetId, new JSONObject(descriptor),
```
Comment (ryanrdoherty) [inline Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java:130]: Could just use setters to assign dataset ID and descriptor here, but probably not much more expensive given the work above.

File: Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java [hunk 3]
```diff
@@ -1,89 +1,153 @@
...
+        nextRow.getNumFilters(), nextRow.getNumComputations(), nextRow.getNumVisualizations());
+    
+    LOG.info("Analysis descriptor after migration: " + out);
```
Comment (ryanrdoherty) [inline Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java:133]: You're dumping the AnalysisRow here; might have a decent toString() but isn't necessarily good for comparing to the "before" log line above.

**Review by ryanrdoherty — CHANGES_REQUESTED** [review]
Really think we need to consider >1 entityId per descriptor.

File: Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java (hunk 1 above)
Comment (dmgaldi) [reply Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java:114]: There's only one entity allowed in ISA simple studies

File: Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java [hunk 4]
```diff
@@ -1,89 +1,153 @@
...
+    }
+
+    final String schema = args.get("--schema");
+    setEntityIdRetriever(new VDIEntityIdRetriever(wdkModel.getAppDb().getDataSource(), schema));
+
+    final File tinyDbFile = new File(args.get("--tinyDb"));
+    readVdiMappingFile(tinyDbFile);
+
+    // Default to dryrun to avoid incidental migrations when testing.
+    _writeToDb = Boolean.parseBoolean(args.getOrDefault("-write", "false"));
```
[the remark this answers is not in the document]
Comment (dmgaldi) [reply Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java]: Yes, fixed

**Review by ryanrdoherty — APPROVED** [review]

**Pull request merged**